pyam/__init__.py

```diff
@@ -10,26 +10,28 @@
 from pyam.iiasa import read_iiasa  # noqa: F401
 from pyam.datareader import read_worldbank  # noqa: F401
 
+from pyam.logging import defer_logging_config
+
 logger = logging.getLogger(__name__)
-logger.addHandler(logging.NullHandler())
 
-# in Jupyter notebooks: disable autoscroll and set logger to info
+# in Jupyter notebooks: disable autoscroll and set-up logging
 try:
-    get_ipython().run_cell_magic(u'javascript', u'',
-                                 u'IPython.OutputArea.prototype._should_scroll = function(lines) { return false; }')
-
-    logger.setLevel(logging.INFO)
-
-    stderr_info_handler = logging.StreamHandler()
-    formatter = logging.Formatter('%(name)s - %(levelname)s: %(message)s')
-    stderr_info_handler.setFormatter(formatter)
-    logger.addHandler(stderr_info_handler)
-
-    log_msg = (
-        "Running in a notebook, setting `{}` logging level to `logging.INFO` "
-        "and adding stderr handler".format(__name__)
-    )
-    logger.info(log_msg)
+    from ipykernel.zmqshell import ZMQInteractiveShell
+    from IPython import get_ipython
+
+    shell = get_ipython()
+    if isinstance(shell, ZMQInteractiveShell):
+        shell.run_cell_magic(
+            u'javascript', u'',
+            u'IPython.OutputArea.prototype._should_scroll = '
+            u'function(lines) { return false; }'
+        )
+
+        log_msg = "Running in a notebook, setting root logging level to INFO"
```

Comment:
I would add here that we are also running basic config for the user

Reply:
I am trying to keep it short! How is

Running in a notebook, setting up a basic logging config at level INFO